src/main/io/serial.h

```diff
@@ -83,7 +83,8 @@ extern const uint32_t baudRates[];
 typedef enum {
     SERIAL_PORT_ALL = -2,
     SERIAL_PORT_NONE = -1,
-    SERIAL_PORT_USART1 = 0,
+    SERIAL_PORT_UART0 = 0,
```

Comment:
Maybe use something like:

```suggestion
#ifdef PLATFORM_TRAIT_HAS_UART0
    SERIAL_PORT_UART0 = 0,
    SERIAL_PORT_UART_FIRST = SERIAL_PORT_UART0,
    SERIAL_PORT_USART1,
#else
    SERIAL_PORT_USART1 = 0,
    SERIAL_PORT_UART_FIRST = SERIAL_PORT_USART1,
#endif
```

Reply:
Even with this - all the rest needs fixing... I think the best option is to remove the use of index from cli scripts and move to names.

The configurator can be versioned to +1, and selections.